pkg/apis/v1/nodeclaim_validation.go

```diff
@@ -149,3 +155,44 @@ func ValidateRequirement(requirement NodeSelectorRequirementWithMinValues) error
 	}
 	return errs
 }
+
+// ValidateWellKnownValues checks if the requirement has well known values.
+// An error will cause a NodePool's Readiness to transition to False.
+// It returns an error if all values are invalid.
+// It returns an error if there are not enough valid values to satisfy min values for a requirement with known values.
+// It logs if invalid values are found but valid values can be used.
+func validateWellKnownValues(ctx context.Context, requirement NodeSelectorRequirementWithMinValues) error {
+	// If the key doesn't have well-known values or the operator is not In, nothing to validate
+	if !WellKnownLabels.Has(requirement.Key) || requirement.Operator != v1.NodeSelectorOpIn {
+		return nil
+	}
+
+	// If the key doesn't have well-known values defined, nothing to validate
+	knownValues, exists := WellKnownValuesForRequirements[requirement.Key]
+	if !exists {
+		return nil
+	}
+
+	values, rejectedValues := lo.FilterReject(requirement.Values, func(val string, _ int) bool {
+		return knownValues.Has(val)
+	})
+
+	// If there are only invalid values, set an error to transition the nodepool's readiness to false
+	if len(values) == 0 {
+		return fmt.Errorf("no valid values found in %v for %s, expected one of: %v, got: %v",
+			requirement.Values, requirement.Key, knownValues, rejectedValues)
+	}
+
+	// If there are valid values, but the minimum number of values is not met, set an error to prevent the nodepool from going ready
+	if requirement.MinValues != nil && len(values) < lo.FromPtr(requirement.MinValues) {
```

Comment:
Is this a validation that we could consider doing in CEL? Or would that be too complicated to compute?

Reply:
I think it's too complicated to be allowed because we have to map through every requirement.